glusterd2/provisioners/provisioners.go

```diff
@@ -0,0 +1,54 @@
+package provisioners
+
+import (
+	"errors"
+
+	"github.com/gluster/glusterd2/plugins/lvmprovisioner"
+)
+
+// Provisioner represents bricks provisioner
+type Provisioner interface {
+	// Register will be called when user registers a device or directory.
+	// PV and VG will be created in case of lvm plugin
+	Register(devpath string) error
+	// AvailableSize returns available size of the device
+	AvailableSize(devpath string) (uint64, uint64, error)
+	// Unregister will be called when device/directory needs to be removed
+	Unregister(devpath string) error
+	// CreateBrick creates the brick volume
+	CreateBrick(devpath, brickid string, size uint64, bufferFactor float64) error
```

Comment:
makes sense, I will merge the functions into one. `MountBrick` is also required as separate since bricks needs to be mounted on glusterd start/node reboot.

```
Register
AvailableSize
Unregister
CreateBrick
MountBrick
UnmountBrick
RemoveBrick
```

`CreateBrick` also mounts the brick. but exported `MountBrick` can be used separately when required.

Reply:
yeah possible to move bricks mount to volume start. With this change, we should only mount the bricks if `volume.State=="Started"` during glusterd2 start/restart